aisdc/attacks/attack_report_formatter.py

```diff
@@ -86,51 +128,86 @@ def __init__(self, report: dict):
                 return True
         return False
 
-    def _rf_min_samples_leaf(self, min_samples_leaf_score):
+    def _tree_min_samples_leaf(self, min_samples_leaf_score):
         if "model_params" in self.report:
             if "min_samples_leaf" in self.report["model_params"]:
                 min_samples_leaf = self.report["model_params"]["min_samples_leaf"]
                 if min_samples_leaf < 5:
                     self.scores.append(min_samples_leaf_score)
                     self.reasons.append("Min samples per leaf < 5")
+                    self.support_rejection.append("Min samples per leaf < 5")
+                else:
+                    self.support_release.append("Min samples per leaf > 5")
 
     def _statistically_significant_auc(
         self, p_val_thresh, mean_auc_thresh, stat_sig_score, mean_auc_score
     ):
         stat_sig_auc = []
-        if "attack_experiment_logger" in self.report:
-            for i in self.report["attack_experiment_logger"]["attack_instance_logger"]:
-                instance = self.report["attack_experiment_logger"][
-                    "attack_instance_logger"
-                ][i]
-                if instance["P_HIGHER_AUC"] < p_val_thresh:
-                    stat_sig_auc.append(instance["AUC"])
-
-            n_instances = len(
-                self.report["attack_experiment_logger"]["attack_instance_logger"]
-            )
-            if (
-                len(stat_sig_auc) / n_instances > 0.1
-            ):  # > 10% of AUC are statistically significant
-                self.scores.append(stat_sig_score)
-                self.reasons.append(">10% AUC are statistically significant")
-
-            if len(stat_sig_auc) > 0:
-                mean = np.mean(np.array(stat_sig_auc))
-                if mean > mean_auc_thresh:
-                    self.scores.append(mean_auc_score)
-                    self.reasons.append("Attack AUC > threshold")
-
-    def process_dict(self) -> dict:
-        self._rf_min_samples_leaf(self.MIN_SAMPLES_LEAF_SCORE)
+        for k in self.report.keys():
+            if isinstance(self.report[k], dict):
+                if "attack_experiment_logger" in self.report[k]:
+                    for i in self.report[k]["attack_experiment_logger"][
+                        "attack_instance_logger"
+                    ]:
+                        instance = self.report[k]["attack_experiment_logger"][
+                            "attack_instance_logger"
+                        ][i]
+                        if instance["P_HIGHER_AUC"] < p_val_thresh:
+                            stat_sig_auc.append(instance["AUC"])
+
+                    n_instances = len(
+                        self.report[k]["attack_experiment_logger"][
+                            "attack_instance_logger"
+                        ]
+                    )
+                    if (
+                        len(stat_sig_auc) / n_instances > 0.1
```

Comment:
Fine for now, but for the next sprint suggest that we read these hard-coded values from a config fie or something so they can be changed on a per-dataset or per-tRE level.
Then I guess the output strings have to be more like; f"when we tested for the metric {metric_name} in more than {metric_tolerance} of our runs the value was statistically significant"